syncer/metrics.go

```diff
@@ -273,23 +273,156 @@ func InitStatusAndMetrics(addr string) {
+func IncrAddedJobsTotalCounter(labels ...string) {
+	addedJobsTotalCounter.WithLabelValues(labels...).Inc()
+}
+
+// IncrFinishedJobsTotalCounter is a wrapper for finishedJobsTotalCounter.
+func IncrFinishedJobsTotalCounter(labels ...string) {
+	finishedJobsTotalCounter.WithLabelValues(labels...).Inc()
+}
+
+// SetQueueSizeGauge is a setter for queueSizeGauge.
+func SetQueueSizeGauge(size float64, labels ...string) {
+	queueSizeGauge.WithLabelValues(labels...).Set(size)
+}
+
+// SetBinlogPosGauge is a setter for binlogPosGauge.
+func SetBinlogPosGauge(size float64, labels ...string) {
+	binlogPosGauge.WithLabelValues(labels...).Set(size)
+}
+
+// SetBinlogFileGauge is a setter for binlogFileGauge.
+func SetBinlogFileGauge(size float64, labels ...string) {
+	binlogPosGauge.WithLabelValues(labels...).Set(size)
+}
+
+// AddSQLRetriesTotalCounter is adder for sqlRetriesTotalCounter.
+func AddSQLRetriesTotalCounter(num float64, labels ...string) {
+	sqlRetriesTotalCounter.WithLabelValues(labels...).Add(num)
+}
+
+// SetTxnHistogram is a setter for txnHistogram.
+func SetTxnHistogram(size float64, labels ...string) {
+	txnHistogram.WithLabelValues(labels...).Observe(size)
+}
+
+// SetQueryHistogram is a setter for queryHistogram.
+func SetQueryHistogram(size float64, labels ...string) {
+	queryHistogram.WithLabelValues(labels...).Observe(size)
+}
+
+// AddSyncerExitWithErrorCounter is adder for syncerExitWithErrorCounter.
+func AddSyncerExitWithErrorCounter(num float64, labels ...string) {
+	syncerExitWithErrorCounter.WithLabelValues(labels...).Add(num)
+}
+
+// SetReplicationLagGauge is a setter for replicationLagGauge.
+func SetReplicationLagGauge(lag float64, labels ...string) {
+	replicationLagGauge.WithLabelValues(labels...).Set(lag)
+}
+
+// SetRemainingTimeGauge is a setter for remainingTimeGauge.
+func SetRemainingTimeGauge(val float64, labels ...string) {
+	remainingTimeGauge.WithLabelValues(labels...).Set(val)
+}
+
+// SetUnsyncedTableGauge is a setter for unsyncedTableGauge.
+func SetUnsyncedTableGauge(lag float64, labels ...string) {
+	unsyncedTableGauge.WithLabelValues(labels...).Set(lag)
+}
+
+// SetShardLockResolvingGauge is a setter for shardLockResolvingGauge.
+func SetShardLockResolvingGauge(lag float64, labels ...string) {
+	shardLockResolvingGauge.WithLabelValues(labels...).Set(lag)
+}
+
+// AddHeartbeatUpdateErrCounter is adder for heartbeatUpdateErrCounter.
+func AddHeartbeatUpdateErrCounter(num float64, labels ...string) {
+	heartbeatUpdateErrCounter.WithLabelValues(labels...).Add(num)
+}
+
+// AddJobCountMetric update addedJobsTotalCounter/finishedJobsTotalCounter metric.
+func AddJobCountMetric(s *Syncer, isFinished bool, queueBucket string, tp opType, n int64) {
+	m := addedJobsTotalCounter
+	if isFinished {
+		s.count.Add(n)
+		m = finishedJobsTotalCounter
+	}
+
+	switch tp {
+	case insert:
+		m.WithLabelValues("insert", s.cfg.Name, queueBucket, s.cfg.SourceID).Add(float64(n))
```

Comment:
"insert" could be retrieved by tp.String()

Reply:
added